server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -290,7 +308,7 @@ public long getParentLimit() {
     public void checkParentLimit(long newBytesReserved, String label) throws CircuitBreakingException {
         final MemoryUsage memoryUsed = memoryUsed(newBytesReserved);
         long parentLimit = this.parentSettings.getLimit();
-        if (memoryUsed.totalUsage > parentLimit) {
+        if (memoryUsed.totalUsage > parentLimit && doubleCheckMemoryUsed(memoryUsed).totalUsage > parentLimit) {
```

Comment:
Maybe the method call can be replaced with `overLimitStrategy.apply(memoryUsed)` and the construction of the OverLimitStrategy will be responsible for handling the behavior of what to do?

Reply:
Good idea, done in 3f61f93f6660b32153047b7fb0393b314510d238